client/src/components/AddVideo.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from "react";
+
+function AddVideo({ onAddVideo }) {
+  const [title, setTitle] = useState("");
+  const [url, setUrl] = useState("");
+  const [Error, setError] = useState("");
+
+  const handleTitleChange = (event) => {
+    setTitle(event.target.value);
+  };
+
+  const handleUrlChange = (event) => {
+    setUrl(event.target.value);
+  };
+  const validUrl = (inputUrl) => {
+    return inputUrl.match(
+      /^(https?:\/\/)?(www\.)?youtube\.com\/watch\?v=[\w-]+(&\S+)?$/
+    );
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setError("Please enter a title"); //title error handling
+    } else if (!validUrl(url)) {
+      setError("Please enter a valid YouTube URL"); //url error handling
+    } else {
+      onAddVideo({ title, url });
+      setTitle("");
+      setUrl("");
```

Comment:
you can use setError("") here as well so we will clear the error if the form submited correctly